### PR TITLE
Blob Sync Test Coverage

### PR DESCRIPTION
**Pull request by ibrandes**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseApiTests.java [hunk 1]
```diff
@@ -856,4 +880,47 @@ public void accept(BlobQueryError nonFatalError) {
...
+    @Test
+    public void syncCopyBaseSimple() {
+        BlockBlobClient bu2 = cc.getBlobClient(generateBlobName()).getBlockBlobClient();
+        String sas = bc.generateSas(new BlobServiceSasSignatureValues(testResourceNamer.now().plusDays(1),
+            new BlobContainerSasPermission().setReadPermission(true)));
+        assertDoesNotThrow(() -> bu2.copyFromUrl(bc.getBlobUrl() + "?" + sas));
+    }
+
+    @Test
+    public void snapshotBase() {
```
Comment (kyleknap) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseApiTests.java:920]: Instead of using methods on the `BlobClientBase`, could we either just move this test to the `BlobApiTest.java` or migrate this test to one (or all) of the specialized append/page/block clients? Mainly I'm hesitant to be testing against the base class directly as it is more of an implementation detail and users won't be instantiating it directly. By moving it to use one of the public classes that customers will use that gives us a bit more safety in case we decide to refactor this base class or override any of the methods in the client subclasses.

**Review by kyleknap — COMMENTED** [review]
Looks good. I just had some questions/suggestions.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceApiTests.java [hunk 2]
```diff
@@ -848,6 +875,11 @@ public void getAccountInfoMin() {
         assertResponseStatusCode(primaryBlobServiceClient.getAccountInfoWithResponse(null, null), 200);
     }
 
+    @Test
+    public void getAccountInfoSimple() {
+        assertDoesNotThrow(() -> primaryBlobServiceClient.getAccountInfo());
```
Comment (kyleknap) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceApiTests.java:880]: Let's change this one to an `assertNotNull` assertion to make sure we get an object returned

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceApiTests.java [hunk 3]
```diff
@@ -658,6 +658,25 @@ public void setPropsMin() {
         }
     }
 
+    private static Stream<Arguments> setAndGetServiceVersionSupplier() {
+        return Stream.of(
+            Arguments.of(BlobServiceVersion.V2020_02_10, BlobServiceVersion.V2020_02_10)//,
+            //Arguments.of(null, BlobServiceVersion.getLatest())
```
Comment (kyleknap) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceApiTests.java:664]: Was this intended to be commented out? I'm also fine just having these as two separate test cases if trying to parameterize and handle a service version of `null` muddies the test case's logic.

**Review by kyleknap — COMMENTED** [review]
Looks good. Just a couple more comments and we should be good.

**Review by ibrahimrabab — APPROVED** [review]
